docs/advanced.rst

```diff
@@ -1018,105 +1136,665 @@ to learn more about the use of this feature.
+| .. literalinclude:: ../code/CodeTester.cpp                 |
+|    :language: c++                                          |
+|    :start-after: //CONF_SERVER_CLIENT_LOCATORS             |
+|    :end-before: //!--                                      |
++------------------------------------------------------------+
+| **XML**                                                    |
++------------------------------------------------------------+
+| .. literalinclude:: ../code/XMLTester.xml                  |
+|    :language: xml                                          |
+|    :start-after: <!-->CONF-SERVER-CLIENT-LOCATORS<-->      |
+|    :end-before: <!--><-->                                  |
++------------------------------------------------------------+
+
+.. _DS_ping_period:
+
+Client ping period
+^^^^^^^^^^^^^^^^^^
+
+As explained :ref:`above <DS_key_concepts>` the *clients* send hailing messages to the *servers* at regular intervals
+(ping period) until they received message reception acknowledgement.
+This period is specified in the member:
+
+.. code-block:: bash
+
+    ParticipantAttributes.rtps.builtin.discovery_config.discoveryServer_client_syncperiod
+
+or the xml tag:
+
+.. code-block:: bash
+
+    dds>profiles>participant>rtps>builtin>discovery_config>clientAnnouncementPeriod
+
++------------------------------------------------------------+
+| **C++**                                                    |
++------------------------------------------------------------+
+| .. literalinclude:: ../code/CodeTester.cpp                 |
+|    :language: c++                                          |
+|    :start-after: //CONF_SERVER_CLIENT_PING                 |
+|    :end-before: //!--                                      |
++------------------------------------------------------------+
+| **XML**                                                    |
++------------------------------------------------------------+
+| .. literalinclude:: ../code/XMLTester.xml                  |
+|    :language: xml                                          |
+|    :start-after: <!-->CONF-SERVER-CLIENT-PING<-->          |
+|    :end-before: <!--><-->                                  |
++------------------------------------------------------------+
+
+.. _DS_match_period:
+
+Server match period
+^^^^^^^^^^^^^^^^^^^
+
+As explained :ref:`above <DS_key_concepts>` the *Servers* received the hailing messages but they don't start at once to
+share publishers or subscribers info with the newcomers.
+They only trigger this process at regular intervals (match period).
+Note that this member is shared with the *client* setup but its name references solely the *client* functionality.
+This period is specified in the member:
+
+.. code-block:: bash
+
+    ParticipantAttributes.rtps.builtin.discovery_config.discoveryServer_client_syncperiod
+
+or the xml tag:
+
+.. code-block:: bash
+
+    dds>profiles>participant>rtps>builtin>discovery_config>clientAnnouncementPeriod
+
```

Comment:
Get rid of the enum and tag hierarchy, as they are implicit on the examples.

Reply:
As above... It's not easy to imply the jigsaw puzzle that the attributes are, directly, from the examples...that's the main reason I put these clues there.

Reply:
Is the xml tag correct? It does not match with the attribute

Reply:
Yes, I'm afraid I wasn't careful enough when implementing the parsing. I didn't realize there would be two mechanism linked with this period: for clients the **ping** and for servers the **match**.